[test-suite] Document the LLVM test-suite benchmark apps

sjoerdmeijer opened the pull request:
There is no documentation or description of the different apps in the LLVM benchmark test-suite and this is a first attempt to document this for the MultiSource apps.

sjoerdmeijer committed:
[test-suite] Document the LLVM test-suite benchmark apps

There is no documentation or description of the different apps in the
LLVM benchmark test-suite and this is a first attempt to document this.

sparker-arm reviewed (commented):
Seems like a good idea to me. Is it worth adding a column to state whether it's suitable for performance measurements? I guess my broader question would be whether that information is still encoded correctly in CMake too...

sjoerdmeijer commented:
> Seems like a good idea to me. Is it worth adding a column to state whether it's suitable for performance measurements? I guess my broader question would be whether that information is still encoded correctly in CMake too...

Thanks, good point.

I have checked that all apps are suitable for perf measurements. I.e., non of the top-level apps/directories are excluded, and only a few apps (benchmark suites) have sub-directories that are excluded when TEST_SUITE_BENCHMARKING_ONLY is not set, for example this is the pattern:

     ...
    add_subdirectory(17-bintr)
    add_subdirectory(18-imp)
    if(NOT TEST_SUITE_BENCHMARKING_ONLY)
        add_subdirectory(15-trie)
    endif()

I have removed these benchmark from the "remark" column if it was included and spelled out to be part of the benchmark suite, and added a sentence that should clarify that all apps are suitable for perf measurements.

sparker-arm reviewed (approved):
Okay, cheers. LGTM

fhahn commented:
Thanks for putting this together!

sjoerdmeijer commented:
> Thanks for putting this together!

Cheers. I will follow up with the SingleSource apps, but wanted to test the water for MultiSource first, and it is a bit of work.

Just as some sort of heads up, I wanted to look at the exact contents of the test-suite first before proposing some changes to it and this is basically the result of that. I will work on a little proposal to add some new ones and changes some existing apps.
